File: app/src/main/java/me/ameriod/lib/mvp/app/person/PersonContract.java
package me.ameriod.lib.mvp.app.person;

import android.support.annotation.NonNull;

import io.reactivex.Observable;
import me.ameriod.lib.mvp.Mvp;
import me.ameriod.lib.mvp.app.models.Person;

public class PersonContract {

    public interface Interactor {

        Observable<Person> getPerson();

    }

    public interface Presenter extends Mvp.Presenter<View> {

        void getPerson();

    }

    public interface View extends Mvp.View {

        void setPerson(@NonNull Person person);
    }
}

File: app/src/main/java/me/ameriod/lib/mvp/app/person/PersonContract.kt
package me.ameriod.lib.mvp.app.person

import io.reactivex.Observable
import me.ameriod.lib.mvp.Mvp
import me.ameriod.lib.mvp.app.models.Person

class PersonContract {

    interface Interactor {

        fun getPerson(): Observable<Person>

    }

    interface Presenter : Mvp.Presenter<View> {

        fun getPerson()

    }

    interface View : Mvp.View {

        fun setPerson(person: Person)
    }
}
